example_datascience/__main__.py

```diff
@@ -1,4 +1,220 @@
 import sys
+import pandas as pd
+import numpy as np
+from pycaret.classification import ClassificationExperiment, create_model, check_fairness
+from fairlearn.metrics import (
+    MetricFrame,
+    selection_rate,
+    false_positive_rate,
+    false_negative_rate,
+    count,
+    demographic_parity_ratio,
+    demographic_parity_difference,
+)
+from sklearn.metrics import accuracy_score, precision_score
+from fairlearn.reductions import DemographicParity, ExponentiatedGradient
+from aif360.metrics import BinaryLabelDatasetMetric
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import StandardScaler
+from sklearn.compose import ColumnTransformer
+from aif360.datasets import StandardDataset
+from data.download_dataset import download_from_kaggle
+import lightgbm as lgb
+from sklearn.pipeline import make_pipeline
+from sklearn.pipeline import Pipeline
+from sklearn.model_selection import cross_val_predict
+import joblib
+
+np.random.seed(69)
+
+
+def exploratory_data_analysis():
+    """
+    This function will do the exploratory data analysis (EDA) on the recruitment dataset of the University of Utrecht.
+    In most project this is done in a Jupyter Notebook as this makes interaction easier, but a slice of the EDA is still
+    implementated to give you an idea.
+    :return: pd.Dataframe containing the data
+    """
+    datapath = "./data/recruitmentdataset-2022-1.3.csv"
+    download_from_kaggle(datapath)
+
+    df = pd.read_csv(datapath, index_col="Id")
+
+    # print(df.head()) # The data is about whether based on some characteristic people are hired or not
+    # print(df.shape) # There are 4000 datapoints with 13 feature columns and 1 label column
+    # print(df[df.duplicated()]) # There are 39 duplicate values, but this is to be expected based on the categorical features
+    # print(df.isna().sum()) # There are no missing values
+    # print(df.describe()) # There are just 3 numerical features, which are the age, university grade, and languages.
+    # print(df.columns)
+    cat_cols = [
+        "gender",
+        "nationality",
+        "sport",
+        "ind-debateclub",
+        "ind-programming_exp",
+        "ind-international_exp",
+        "ind-entrepeneur_exp",
+        "ind-exact_study",
+        "company",
```

Comment:
The dataset contains information about 4 different companies, maybe it could be useful to also split the data into 4 subsets of these companies to find out whether there are different biases present across different companies

Reply:
valid point!

Reply:
As right now I would use it as feature, which doesn't make so much sense